### ISSUE TITLE
documentation on environment variables is lacking

### ISSUE THREAD

**Issue by Be-ing** [rust-lang/cc-rs#920]
Currently, the documentation on what environment variables this crate reads and what effects they have is lacking. For example, the [try_flags_from_environment documentation](https://docs.rs/cc/latest/cc/struct.Build.html#method.try_flags_from_environment) merely says "Normally the cc crate will consult with the standard set of environment variables (such as CFLAGS and CXXFLAGS) to construct the compiler invocation." cc-rs uses more than `CFLAGS` and `CXXFLAGS`. How to use `CC`/`CXX` is not documented, nor how to use target-specific environment variables, nor how `CC`/`CXX` can specify compiler caches and compiler flags, nor how `RUSTC_WRAPPER` can be used for C & C++ compiler caching.

**Comment by NobodyXu** [comment]
There are a few more from https://github.com/rust-lang/cc-rs/blob/8cf5455c22b297f95dcbbfc06e1dfd68c34e19e9/src/lib.rs#L3238 :

 - `TARGET`: set the default target if user does not specify one
 - `HOST`
 - `OPT_LEVEL`
 - `DEBUG`

**Comment by NobodyXu** [comment]
Also

 - `CRATE_CC_NO_DEFAULTS` https://github.com/rust-lang/cc-rs/blob/8cf5455c22b297f95dcbbfc06e1dfd68c34e19e9/src/lib.rs#L1628
 - `CARGO_CFG_TARGET_FEATURE` https://github.com/rust-lang/cc-rs/blob/8cf5455c22b297f95dcbbfc06e1dfd68c34e19e9/src/lib.rs#L1713
 - `RUSTC_LINKER` https://github.com/rust-lang/cc-rs/blob/8cf5455c22b297f95dcbbfc06e1dfd68c34e19e9/src/lib.rs#L3080
 - `CROSS_COMPILE` https://github.com/rust-lang/cc-rs/blob/8cf5455c22b297f95dcbbfc06e1dfd68c34e19e9/src/lib.rs#L3083

**Comment by NobodyXu** [comment]
There are also prefixed version of `CFLAGS`/`CXXFLAGS`

https://github.com/rust-lang/cc-rs/blob/8cf5455c22b297f95dcbbfc06e1dfd68c34e19e9/src/lib.rs#L2180

**Comment by NobodyXu** [comment]
Also prefixed  in `CC` and `CXX`

https://github.com/rust-lang/cc-rs/blob/8cf5455c22b297f95dcbbfc06e1dfd68c34e19e9/src/lib.rs#L2558

https://github.com/rust-lang/cc-rs/blob/8cf5455c22b297f95dcbbfc06e1dfd68c34e19e9/src/lib.rs#L2755

**Comment by NobodyXu** [comment]
Prefixed `AR` and `RANLIB`

https://github.com/rust-lang/cc-rs/blob/8cf5455c22b297f95dcbbfc06e1dfd68c34e19e9/src/lib.rs#L2949C17-L2949C25

**Comment by Be-ing** [comment]
Oh, there is some [documentation in README.md](https://github.com/rust-lang/cc-rs#external-configuration-via-environment-variables), but not on https://docs.rs. I think that should be moved to the crate documentation in lib.rs so it shows up on https://docs.rs. I expect to find crate usage documentation on https://docs.rs, not a README.md in the source repository.

**Comment by NobodyXu** [comment]
Maybe we could `#[doc = include_str!("../../README.md")]
` to include them in docs.rs/cc?

**Comment by Be-ing** [comment]
I think it would be simpler to just link to https://docs.rs from README.md and keep the documentation in one place (lib.rs)

**Comment by NobodyXu** [comment]
Well, if we add new env to main, then anybody who checks the doc in README would use it while it doesn't work in practice.

**Comment by Be-ing** [comment]
Yes, I've run into that problem with another Rust library. I got confused because the README.md documented some new features that were not available in the latest release at the time.

**Comment by NobodyXu** [comment]
Yeah, that's why I think `#[doc = include_str!("../../README.md")]` makes more sense, since it would guarantee the consistency

**Comment by Be-ing** [comment]
`#[doc = include_str!("../../README.md")]` would keep README.md and docs.rs consistent. However, that leaves the possibility of someone going to https://github.com/rust-lang/cc-rs and getting confused by new information in README.md that hasn't been released yet.

**Comment by NobodyXu** [comment]
Yeah that's true, my other project simply leaves a link to latest doc on crates.io/crates/cargo-binstall to maintain consistency
